### DISCUSSION TITLE
Please include mysqldump

### DISCUSSION THREAD

**Discussion by DKeppi** [shinsenter/php#73] [Ideas]
Hello 

I have an Laravel Application which makes a backup on its own when updating.
Sadly i get a lot of errors due to the missing command mysqldump

Could you please include the command in the image.

Thanks a lot and best regards
DKeppi

**Comment by shinsenter** [comment]
Hi @DKeppi 

The Docker images we provide are primarily focused on creating an environment for PHP and web server applications written in pure PHP. To keep the Docker image size optimized, we do not pre-install tools that may not be needed by every application.

If you need additional tools like `mysqldump`, you can install them using `apk` (on Alpine Linux) or `apt` (on Debian) commands in your Dockerfile.

Here’s an example:

**For Alpine Linux:**

```Dockerfile
FROM shinsenter/php:latest-alpine

# Install mysqldump
RUN apk --no-cache add mysql-client

# Your other commands
...
```

**For Debian:**

```Dockerfile
FROM shinsenter/php:latest

# Install mysqldump
RUN apt-get update && apt-get install -y mysql-client && rm -rf /var/lib/apt/lists/*

# Your other commands
...
```

To build a Docker image from the above Dockerfile, navigate to the directory containing your Dockerfile and run the following command:

```shell
docker build -t 'shinsenter/php:latest-with-mysqldump' .
```

I hope this helps resolve the issue.

**Comment by DKeppi** [comment]
Sadly that doesnt work:

```
apt-get update && apt-get install -y mysql-client && rm -rf /var/lib/apt/lists/*
Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
Get:4 http://deb.debian.org/debian bookworm/main amd64 Packages [8788 kB]
Get:5 http://deb.debian.org/debian bookworm-updates/main amd64 Packages [13.8 kB]
Get:6 http://deb.debian.org/debian-security bookworm-security/main amd64 Packages [168 kB]
Fetched 9224 kB in 3s (3436 kB/s)                         
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package mysql-client is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'mysql-client' has no installation candidate
```

**Comment by DKeppi** [comment]
`apt-get update && apt-get install -y default-mysql-client && rm -rf /var/lib/apt/lists/*` worked for me
But i installed it in the running container, i dont want to build my own one!

**Reply by shinsenter** [reply]
Yeah, you can definitely choose the method that suits your specific project.

Have a nice day!

**Reply by DKeppi** [reply]
You too thx!
